erpnext/accounts/doctype/payment_request/payment_request.py

```
@@ -204,18 +225,160 @@ def before_submit(self):
 	def on_submit(self):
 		self.update_reference_advance_payment_status()
 
+	def _process_v2_gateway(self):
+		"""Process payment using the new PaymentController interface (v2 gateways)."""
+		tx_data = self.get_tx_data()
+		with payment_app_import_guard():
+			from payments.controllers import PaymentController
+
+			try:
+				_controller, psl_name = PaymentController.initiate(tx_data, self.payment_gateway)
+			except Exception as e:
```

Ruff (0.15.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)